Tribal crafting expansion

Flakkyboo opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds some much needed love to the Tribal crafting tree by adding some new storage options and some new weapons and equipment to make the give people more options later in the round.
overhauls the entire bone sprite set

Adds two new types of bows. 
Adds four new arrow types.  
Adds a new sword and shield.
Adds a new bola.
Tweaked the blowgun.
Adds a new storage option in the arrow quiver.
new sprites for existing bone items (armor, axe spear)
Adds code to make an existing tribal belt to be craftable.
cleans up some code with cotton as currently cloth is not considered a subtype of cotton.
refactor of crafting as introduced in https://github.com/tgstation/tgstation/pull/48451 and the subsequent fix #48477

Sprites generously provided by @SomeAngryMiner and @thatguythere03 

Pulls inspiration and some code from the following
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11098
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10993
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11201
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11401
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11516
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11172
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11367

## Why It's Good For The Game

The tribal crafting tree is in much need of a expansion its typically only used by ashwalkers and the occasional person looking to dry tobacco or brew drinks. 

This will hopefully give the players that use the tribal crafting more options in terms of weaponry and storage with a clear progression of power as round progresses. New fancy sprites to give tribal crafting a new fresh feel.

![image](https://user-images.githubusercontent.com/63934530/82841726-06ef8200-9ecf-11ea-9ce9-798c97c347ff.png)
Above: Examples of the new sprites and items

![image](https://user-images.githubusercontent.com/63934530/82842010-fe4b7b80-9ecf-11ea-9395-3ee9861b9895.png)
Above: How the new items look when worn and some new storage options

![image](https://user-images.githubusercontent.com/63934530/82842267-c55fd680-9ed0-11ea-8d84-59bca2f40e61.png)
Above: The bow will work in 3 steps. you must load the arrow, then draw the arrow back, then fire.



## Changelog
:cl:
add: Added a Goliath shield and it takes less damage from fauna.
add: Added a new Watcher bola, slightly better than the standard bola.
add: Added a new one handed bone sword.
add: Added a new belt quiver to carry arrows.
add: Two new craftable bow types "pipe bow" and "bone bow".
add: 4 new arrow types wood, ash, bone and bronze.
add: New export prices for bows, arrows.
add: New bounty for a bow.
add: New variation of bone armor (heavy bone armor)
add: Cotton seeds in askwalker nest for bowcrafting. 
add: QOL cultivator for ashwalkers added to nest.
tweak: blowgun no longer needs a firing pin and can be used by anyone
soundadd: New sounds for drawing the arrow and firing.
add: New weapon and ammo type for bows.
fix: Added a crafting recipe for the primal explorers belt.
add: A new material Silk which can be made from cloth on the loom. (needed in bowcrafting)
imageadd: Added Sprites for the new items and materials.
imageadd: New sprites for existing bone equipment.
refactor: Cloth is now a subtype of cotton and coded in descending order of creation.
crafting recipes are updated to reflect this.
Cotton>cloth>silk
refactor: Crafting refactor 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Flakkyboo committed:
sword,bola,shield,belt

Adds a new shield,bola,belt,sword and recipes.
waiting on inhand sprites for shield and bola

Flakkyboo committed:
Refractoring of sheet_types (cotton)

refactored the Code related to cotton.
cloth is now a subtype of cotton
silk can be made from cloth on a loom which is needed in the bow making process.

zeskorion commented on code/game/objects/items/shields.dm:
```diff
@@ -44,7 +45,10 @@
 		else if(isliving(hitby))
 			var/mob/living/L = hitby
-			attackforce = (damage * 2)//simplemobs have an advantage here because of how much these blocking mechanics put them at a disadvantage
+			if(hunter == TRUE)
```
make this a bitflag, perhaps

Flakkyboo replied:
ill look into how bitflags work sure to see

Flakkyboo replied:
Change implemented

zeskorion commented on code/game/objects/items/weaponry.dm:
```diff
@@ -216,6 +216,28 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	block_power = 15
+	block_level = 1
```
i dunno...

Flakkyboo replied:
too high of a block power?
I was basing it on the block power of the bone spear which was 25 and knocked 10 off that.

Flakkyboo replied:
Removed "Block_level = 1"

Flakkyboo commented:
going to be adding the bow framework today.
its going to take a little time

Flakkyboo committed:
Sword and shield changes

commit changes recommended by review

Flakkyboo committed:
Bows

Adds supporting code for bow creation and firing.
sprites/sounds added for bow,quiver
4 arrow types are coded with sprites added

Flakkyboo commented:
Bows are now implemented and tested.
waiting on inhand sprites for shield and bola.
need to add cotton seeds to ashwalker nest for bow creation

Flakkyboo committed:
Adds cotton to Nest

Adds cotton to ashwalker nest so they can make bows.
QOL add to give them a cultivator also

Flakkyboo committed:
Revert "Adds cotton to Nest"

This reverts commit 536295b32335f59ed145092ba9270a38d9d9db7d.

Flakkyboo commented:
No idea what's causing the merge conflict

Flakkyboo committed:
Merge branch 'master' into Tribal_crafting_expansion

Flakkyboo committed:
Resolved merge conflict

re-added cotton and QOL cultivator to ashwalker den.
Merge conflict resolved hopefully

Flakkyboo committed:
Travis has issues

This should resolve the issue that Travis had with compiling maps. cloth was not spawning on maps due to filepath change caused by previous commit

Flakkyboo commented:
ah i fucked up on the sprites.

Flakkyboo committed:
I fucked up some sprites

I fucked up when i merged the master into the branch and inadvertently reverted the beegun and medical bag pack sprites. its been fixed promise

Flakkyboo commented:
That will be my last commit until reviewed

zeskorion commented on code/game/objects/items/weaponry.dm:
```diff
@@ -216,6 +216,27 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	block_power = 15
+	block_upgrade_walk = 1
```
being a subtype of claymore, this only needs the block level of 0 specified, as well as block power (which can be a good bit higher now that you removing passive block. Armor, integrity, sharpness, and flags are unecessary

Flakkyboo replied:
Done

zeskorion commented on code/modules/projectiles/gun.dm:
```diff
@@ -46,7 +46,7 @@
 
 	var/obj/item/firing_pin/pin = /obj/item/firing_pin //standard firing pin for most guns
-
+	var/no_pin_required = FALSE //whether the gun can be fired without a pin
```
I'm not sure, but this may break some shit.

zeskorion commented on code/modules/projectiles/guns/ballistic/bow.dm:
```diff
@@ -0,0 +1,65 @@
+	casing_ejector = FALSE
+	internal_magazine = TRUE
+	pin = null
+	no_pin_required = TRUE
```
this may break some shit

Flakkyboo replied:
Ive tested every gun by firing them and then removing their pin and trying to fire again.
no gun that was not a debug blastcannon fails to fire without a pin.
hopefully it doesn't break other stuff but for now it seems like were okay.

Fikou replied:
you should make the bamboo syringe gun thing not require a pin

zeskorion commented on code/modules/projectiles/guns/ballistic/bow.dm:
```diff
@@ -0,0 +1,65 @@
+	desc = "A crude projectile weapon made from silk string, pipe and lots of bending."
+	icon_state = "pipebow"
+	item_state = "pipebow"
+	force = 0
```
set this to the force of a pipe

Flakkyboo replied:
done

zeskorion commented on code/modules/projectiles/guns/ballistic/bow.dm:
```diff
@@ -0,0 +1,65 @@
+		user.put_in_hands(AC)
+		chambered = null
+		to_chat(user, "<span class='notice'>You gently release the bowstring, removing the arrow.</span>")
+	else if (get_ammo())
```
maybe a do_mob to pull back the string, which goes back to taught if not in an active hand

Flakkyboo replied:
working on this

zeskorion commented on code/modules/projectiles/projectile/reusable/arrow.dm:
```diff
@@ -0,0 +1,25 @@
+/obj/item/projectile/bullet/reusable/arrow/bone //AP for ashwalkers
+	name = "Bone arrow"
+	desc = "An arrow made from bone and sinew."
+	damage = 35
```
I think this is a bit excessive. should be the same as the ashen arrow at most if you are gonna stack that AP on. Miners don't have as much Bullet armor as they have Melee

Flakkyboo replied:
yeah agreed after some tests on different armored targets.
ill make some tweaks for balance

Flakkyboo replied:
tweaked

zeskorion commented on code/modules/projectiles/projectile/reusable/arrow.dm:
```diff
@@ -0,0 +1,25 @@
+/obj/item/projectile/bullet/reusable/arrow/bronze
+	name = "Bronze arrow"
+	desc = "Bronze tipped arrow"
+	armour_penetration = 10
```
why do bronze arrows suck so much

Flakkyboo replied:
they are a bit better now

Flakkyboo committed:
Easy balance changes 

changed some balance according to Review notes

Flakkyboo commented:
Okay now for the hard stuff.
 1. testing "no_pin_required" variable to make sure it doesn't break guns. (tested. no guns that are accessible and not the bow seem to work)

 2. fixing pin removal recipe not being true.

Flakkyboo committed:
do_mob pass 1

implemented a do_mob for releasing a notched arrow so its not an immediate thing.

removed the recipe that was supposed to prevent bows from being valid for pin removal recipe as it broke the pin recipe.

bows are valid for the pin removal recipe but are not affected due to having the no_pin_required variable

Flakkyboo committed:
Bells and whistles

Fixed the do_mob in the bow
added some export values and a bounty for the bone bow and arrows

Flakkyboo committed:
Merge branch 'master' into Tribal_crafting_expansion

Flakkyboo commented:
Merge conflict caused by clothing repath is resolved.
waiting for review before i commit more

Flakkyboo committed:
do_mob pass 2

used a do_mob to release the arrow like a big unga.

moved the do_mob to drawing the bow instead because it takes effort to draw the bow

Flakkyboo committed:
Merge remote-tracking branch 'Flakkyboo/Tribal_crafting_expansion' into Tribal_crafting_expansion

Flakkyboo commented:
diff bot broke

zeskorion reviewed (dismissed):
Even though the bow does not stay strung when not in active hand, making it more like a crossbow, i'll let that slide

Flakkyboo committed:
Bola inahnds

Flakkyboo commented:
Waiting on the final shield inhands before I open this.
New bone armor, bone axe, bone spear sprites incoming.
Need to look into making bone armor similar in function to ash drake armor.
travis is being unfair

Flakkyboo committed:
bone sprites overhaul

new bone armor sprite
heavy bone armor sprite
new bone spear sprite
new bone axe sprite
inhand sprites

Flakkyboo commented:
need back sprites 
working on conflicts

Flakkyboo committed:
Merge branch 'master' into Tribal_crafting_expansion

Flakkyboo committed:
conflict resolved 

fixed some shield sprite errors

Flakkyboo commented:
conflict resolved now all thats left is the back sprites.
i will leave open but do not merge as it

Flakkyboo committed:
slight sprite updates

both bone armor sets now have different names

Flakkyboo committed:
Sprite fixes

Fikou commented on code/modules/projectiles/guns/ballistic/bow.dm:
```diff
@@ -0,0 +1,67 @@
+	internal_magazine = TRUE
+	pin = null
+	no_pin_required = TRUE
+	trigger_guard = TRIGGER_GUARD_NONE //so ashwalkers can use it
```
```suggestion
	trigger_guard = TRIGGER_GUARD_ALLOW_ALL //so ashwalkers can use it
```

Flakkyboo replied:
that can be done thank you for the suggestion

Flakkyboo committed:
Back sprites

back sprites for the shield, spear, axe

Flakkyboo committed:
Suggestion from review 

tested and committed change as noted in review

Flakkyboo commented:
ready to be put to vote.
may work on backsprites for bows if i get the time in the week

Flakkyboo commented:
That's odd I can't see this comment.
I agree that it should and it probably should have a trigger guard compatible with everything

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
From: Fikou <notifications@github.com>
Sent: Tuesday, May 26, 2020 9:22:06 PM
To: BeeStation/BeeStation-Hornet <BeeStation-Hornet@noreply.github.com>
Cc: Flakkyboo <eric_murray101@hotmail.com>; Author <author@noreply.github.com>
Subject: Re: [BeeStation/BeeStation-Hornet] Tribal crafting expansion (#1802)


@Fikou commented on this pull request.

________________________________

In code/modules/projectiles/guns/ballistic/bow.dm<https://github.com/BeeStation/BeeStation-Hornet/pull/1802#discussion_r430685287>:

> +/obj/item/gun/ballistic/bow
+       name = "wooden bow"
+       desc = "some sort of primitive projectile weapon. used to fire arrows."
+       icon_state = "bow"
+       item_state = "bow"
+       w_class = WEIGHT_CLASS_BULKY
+       weapon_weight = WEAPON_HEAVY //need both hands to fire
+       force = 5
+       mag_type = /obj/item/ammo_box/magazine/internal/bow
+       fire_sound = 'sound/weapons/bowfire.ogg'
+       slot_flags = ITEM_SLOT_BACK
+       item_flags = NONE
+       casing_ejector = FALSE
+       internal_magazine = TRUE
+       pin = null
+       no_pin_required = TRUE


you should make the bamboo syringe gun thing not require a pin

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/BeeStation/BeeStation-Hornet/pull/1802#discussion_r430685287>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/APHZAQUTSZTOVFQ3LFU3RODRTQQG5ANCNFSM4NA7PM4A>.

Flakkyboo committed:
backsprites, bow and blowgun teaks

backsprites for the bows
bows now require permits
blowguns no longer need a firing pin to fire and everyone can use it

Flakkyboo commented:
okay im done now no more commits im happy where it is.
azlan hated the bow sprite now theres a new one

Flakkyboo committed:
Azlan is a god

a perfectionist god

Flakkyboo committed:
readd shield change

one of the merges when i was resolving conflicts removed the change that made the new shield better against fauna.

its now re-added

Flakkyboo commented:
not sure what caused the bot to fail

Flakkyboo committed:
Merge branch 'master' into Tribal_crafting_expansion

Flakkyboo committed:
Merge conflict resolved

Flakkyboo committed:
shield balance changes

reduced the block power of shield

zeskorion reviewed (approved):
we'll see how this leaves ashwalkers, but seems fine at the moment.
